Tools/LocomotionTool/LocomotionTool.cs

```diff
@@ -0,0 +1,559 @@
+
+								var thumb = m_LocomotionInput.thumb;
+								var thumbHeld = thumb.isHeld;
+								if (thumbHeld)
+									consumeControl(thumb);
+
+								var otherThumb = otherLocomotionInput.thumb;
+								var otherThumbHeld = otherThumb.isHeld;
+								if (otherThumbHeld)
+									consumeControl(otherThumb);
+
+								// Press both thumb buttons to reset scale
+								if (thumbHeld && otherThumbHeld)
+								{
+									m_AllowScaling = false;
+
+									rayToRay = otherRayOrigin.position - rayOrigin.position;
+									midPoint = rayOrigin.position + rayToRay * 0.5f;
+									var currOffset = midPoint - cameraRig.position;
+
+									cameraRig.position = midPoint - currOffset / currentScale;
+									cameraRig.rotation = Quaternion.AngleAxis(m_StartYaw, Vector3.up);
+
+									ResetViewerScale();
+								}
+
+								var trigger = m_LocomotionInput.trigger;
+								var triggerHeld = trigger.isHeld;
+								if (triggerHeld)
+									consumeControl(trigger);
+
+								var otherTrigger = otherLocomotionInput.trigger;
+								var otherTriggerHeld = otherTrigger.isHeld;
+								if (otherTriggerHeld)
+									consumeControl(otherTrigger);
+
+								// Press both triggers to reset to origin
+								if (triggerHeld && otherTriggerHeld)
+								{
+									m_AllowScaling = false;
+
+									cameraRig.position = Vector3.up * VRView.HeadHeight;
+									cameraRig.rotation = Quaternion.identity;
+
+									ResetViewerScale();
+								}
+
+								if (m_AllowScaling)
+								{
+									var yawSign = Mathf.Sign(Vector3.Dot(Quaternion.AngleAxis(90, Vector3.down) * m_StartDirection, rayToRay));
+									var currentYaw = m_StartYaw + Vector3.Angle(m_StartDirection, rayToRay) * yawSign;
+									var currentRotation = Quaternion.AngleAxis(currentYaw, Vector3.up);
+									midPoint = currentRotation * midPoint * currentScale;
+
+									cameraRig.position = m_StartPosition + m_StartMidPoint - midPoint;
+									cameraRig.localScale = Vector3.one * currentScale;
+									cameraRig.rotation = currentRotation;
+
+									m_MainCamera.nearClipPlane = m_OriginalNearClipPlane * currentScale;
+									m_MainCamera.farClipPlane = m_OriginalFarClipPlane * currentScale;
+
+									Shader.SetGlobalFloat(k_WorldScaleProperty, 1f / currentScale);
+								}
+								break;
+							}
+						}
+
+						if (!otherGripHeld)
+							CancelScale();
+					}
+				}
+				else
+				{
+					CancelScale();
+				}
+			}
+		}
+
+		void ResetViewerScale()
+		{
+			cameraRig.localScale = Vector3.one;
+			m_MainCamera.nearClipPlane = m_OriginalNearClipPlane;
+			m_MainCamera.farClipPlane = m_OriginalFarClipPlane;
+
+			if (m_ViewerScaleVisuals)
+				ObjectUtils.Destroy(m_ViewerScaleVisuals.gameObject);
+		}
+
+		void CreateViewerScaleVisuals(Transform leftHand, Transform rightHand)
+		{
+			m_ViewerScaleVisuals = ObjectUtils.Instantiate(m_ViewerScaleVisualsPrefab, cameraRig, false).GetComponent<ViewerScaleVisuals>();
```

Comment:
Probably want to cache/show/hide these visuals rather than remaking them all the time